ast/compile.go

```diff
@@ -894,6 +894,40 @@ func parseSchema(schema interface{}) (types.Type, error) {
 		return parseSchema(subSchema.RefSchema)
 	}
 
+	// Handle anyOf
+	if subSchema.AnyOf != nil {
+		var orType types.Type
+
+		// If there is a core schema, find its type first
+		if subSchema.Types.IsTyped() {
+			copySchema := *subSchema
+			copySchemaRef := &copySchema
+			copySchemaRef.AnyOf = nil
+			coreType, err := parseSchema(copySchemaRef)
+			if err != nil {
+				return nil, fmt.Errorf("unexpected schema type %v: %w", subSchema, err)
+			}
+
+			// Only add Object type with static props to orType
+			if objType, isObj := coreType.(*types.Object); isObj {
```

Comment:
[nit] It's just a style question, I guess, but I think it would be slightly more idiomatic like this
```suggestion
			if objType, ok := coreType.(*types.Object); ok {
```

..since the type assertion itself is already telling us what "ok" means here -- "it's an object".

Reply:
Done (651e9803e56d4937c0e1a1b2afcc6d0aa4c6342d)